### PR TITLE
Add 6.0 changelog

### PR DESCRIPTION
**Pull request by scotttrinh**

### CODE REVIEW

File: docs/changelog/6_x.rst [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+your project. Pre-release versions are not considered stable and will not be
+automatically suggested:
+
+.. code-block:: bash
+
+  $ edgedb project init --server-version 6.0-beta.1
+
+
+Upgrading
+=========
```
Comment (1st1) [inline docs/changelog/6_x.rst:16]: I'd collapse this section, we have an example somewhere

Comment (scotttrinh) [reply docs/changelog/6_x.rst:16]: Just during the beta period?

File: docs/changelog/6_x.rst [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+
+We aim for this version to be the last requiring a full dump and restore
+process for major version upgrades. We understand that dump-and-restore is
+disruptive, so enabling in-place upgrades will make it easier for teams to
+upgrade more frequently.
+
+Query performance observability
+-------------------------------
+
+We now store statistics about query performance.
```
Comment (1st1) [inline docs/changelog/6_x.rst:124]: Can we add a screenshot here? Also would be great to expand a bit, like that client APIs now have `withQueryTag` method to apply custom perf tags to your queries in your code

File: docs/changelog/6_x.rst [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+
+``ext::postgis``
+----------------
+
+We've added support for the popular PostGIS extension for PostgreSQL. This
+extension adds support for geographic objects and spatial data types.
```
Comment (1st1) [inline docs/changelog/6_x.rst:130]: Same, a screenshot here would be amazing

File: docs/changelog/6_x.rst [hunk 4]
```diff
@@ -0,0 +1,219 @@
...
+
+``std::net``
+------------
+
+We've introduced a new standard library module for sending network requests,
+initially supporting HTTP. This module schedules asynchronous requests and
+allows you to poll for responses.
```
Comment (1st1) [inline docs/changelog/6_x.rst:137]: Maybe an edgeql example?

File: docs/changelog/6_x.rst (hunk 2 above)
Comment (scotttrinh) [reply docs/changelog/6_x.rst:124]: The UI for this isn't available in the beta yet. Do you think it still makes sense to show a screenshot now?

**Review by 1st1 — APPROVED** [review]
good first pass

**Pull request merged**